plugins/modules/content_view_filter.py

```diff
@@ -22,149 +22,126 @@
 DOCUMENTATION = '''
 ---
 module: content_view_filter
-version_added: 1.0.0
-short_description: Manage Content View Filters
+version_added: 3.9.0
+short_description: Manage content view filters
 description:
-    - Create and manage content View filters
-author: "Sean O'Keeffe (@sean797)"
+    - Create and manage content view filters
+author:
+    - "Paul Armstrong (@parmstro)"
 options:
-  architecture:
-    description:
-      - package architecture
-    type: str
-  name:
+  content_view:
     description:
-      - Name of the Content View Filter
-    type: str
+      - name of the content view to add the filter to
     required: true
+    type: str
   description:
     description:
-      - Description of the Content View Filter
+      - set the description of the content view filter
     type: str
-  content_view:
+  inclusion:
     description:
-      - Name of the content view
-    required: true
-    type: str
-  filter_state:
+      - true creates an include filter
+      - false creates an exclude filter
+    default: false
+    type: bool
+  name:
     description:
-      - State of the content view filter
-    default: present
-    choices:
-      - present
-      - absent
+      - set the name of the content view filter
     type: str
+    required: true
+  original_packages:
+    description:
+      - include all packages with no errata
+      - only valid for rpm type filters
+    type: bool
+  original_module_streams:
+    description:
+      - include all module streams with no errata
+      - only valid for modulemd type filters
+    type: bool
   repositories:
     description:
-      - List of repositories that include name and product
+      - list of repositories that the filter applies to including name and product
       - An empty Array means all current and future repositories
     default: []
     type: list
     elements: dict
-  rule_state:
+  state:
     description:
-      - State of the content view filter rule
-    default: present
+      - set the presence or absence of the content view filter
     choices:
       - present
       - absent
     type: str
-  filter_type:
+    default: present
+  type:
     description:
-      - Content view filter type
+      - the type of content to apply the filter to
+    aliases:
+      - filter_type
     required: true
     choices:
       - rpm
+      - modulemd
       - package_group
       - erratum
+      - erratum_id
+      - erratum_date
       - docker
     type: str
-  rule_name:
```

Comment:
the whole removal here is because we now have a dedicated rule module, right? I think I would prefer to keep those parameters *for now* (and mark them as deprecated), as that will allow users who have working setups to migrate gradually vs being forced to do now.

what do you think @mdellweg @ehelms

Reply:
:+1: To leaving them around deprecated for a bit

Reply:
+1 for compatibility. We would need to make a breaking change release otherwise. Deprecating first is the nice thing to do.